src/service.rs

```diff
@@ -424,14 +836,410 @@ impl Service {
                         self.send_ping(enr);
                     }
                 }
+                Some(Ok((active_topic, active_ticket))) = self.tickets.next() => {
+                    // When the ticket time expires a new REGTOPIC request is automatically sent to the
+                    // ticket issuer and the registration attempt stays in the [`RegistrationState::Ticket`]
+                    // from sending the first REGTOPIC request to this contact for this topic.
+                    self.reg_topic_request(active_ticket.contact(), active_topic.topic().clone(), RequestTicket::RemotelyIssued(active_ticket.ticket()));
+                }
+                Some(topic_query_progress) = self.active_topic_queries.next() => {
+                    match topic_query_progress {
+                        TopicQueryState::Finished(topic_hash) | TopicQueryState::TimedOut(topic_hash) => {
+                            if let Some(query) = self.active_topic_queries.queries.remove(&topic_hash) {
+                                if let Some(callback) = query.callback {
+                                    if callback.send(Ok(query.results.into_values().collect::<Vec<_>>())).is_err() {
+                                        warn!("Callback dropped for topic query {}. Results dropped", topic_hash);
+                                    }
+                                }
+                            }
+                        }
+                        TopicQueryState::Unsatisfied(topic_hash) => self.send_topic_queries(topic_hash, None),
+                    }
+                }
+                _ = registration_interval.tick() => {
+                    trace!("New registration interval, {}/{} topics to publish", topics_to_reg_iter.clone().count(), self.registration_attempts.len());
+                    let mut sent_regtopics = 0;
+                    let mut topic_item = topics_to_reg_iter.next();
+                    while let Some((topic, _topic_hash)) = topic_item {
+                        trace!("Publishing topic {} with hash {}", topic, topic.hash());
+                        topic_item = topics_to_reg_iter.next();
```

Comment:
I couldn't find a particular reason to do this using an extra variable. This might be a more idiomatic way of doing the same
```suggestion
                    while let Some((topic, _topic_hash)) = topics_to_reg_iter.next() {
                        trace!("Publishing topic {} with hash {}", topic, topic.hash());
```